spec/message.abnf

```diff
@@ -1,18 +1,30 @@
-message = pattern / complex-message
+message = simple-message / complex-message
 
-complex-message = "{{" [s] *(declaration [s]) body [s] "}}"
+simple-message = [almost-keyword]
+               / ((almost-keyword not-keyword / text-start / expression) pattern)
+complex-message = *(declaration [s]) body
 
-declaration = input-declaration / local-declaration
+declaration = input-declaration / local-declaration / reserved-declaration
 input-declaration = input [s] variable-expression
 local-declaration = local s variable [s] "=" [s] expression
+reserved-declaration = "." 4*(%x61-7A) reserved-body 1*expression
```

Comment:
This has two potential problems:
1. It conflicts with the `input-declaration` and `local-declaration` rules. Is that ok, or do we need the ABNF to be that meticulous?
2. This is requiring every declaration to end with at least one expression. Is that too restricting? If so, can ABNF somehow do a non-capturing lookahead for .keywords as a rule terminator?

Reply:
This is too opinionated about future keyword use. You don't need to define a reserved-declaration. Just let a future keyword define what it needs.

Reply:
Doing so would break our self-imposed stability policy. Specifically, this part: https://github.com/unicode-org/message-format-wg/blob/85e16398b07a91d8b1a16eccdec474af4e6198cc/spec/README.md?plain=1#L100-L104

If we don't introduce a `reserved-declaration`, then we won't be able to define any further declarations or keyword operations without calling it MF3 or something.

Reply:
This is a good callout, but I would remove the `-declaration` from the name because it might not be a declaration but instead some other kind of syntax. I would also not do `4*`. I would suggest:

```suggestion
reserved-syntax = "." %s*[a-z] [(s reserved-body)] 1*expression
```

Reply:
The latest draft has https://github.com/unicode-org/message-format-wg/blob/915cbb7f096815d4c6d908fb8bfac162dfe86b1b/spec/message.abnf#L12

as the keyword also matches `reserved-body`. I'll update the name and then resolve this comment.